--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,11 @@ elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(LLVMFlang)$")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0")
   set(fortran_d_flags "-fdefault-real-8")
   set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(NVHPC)$")
+  set(CMAKE_Fortran_FLAGS "-g -cpp ${CMAKE_Fortran_FLAGS}")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0")
+  set(fortran_d_flags "-r8")
+  set(fortran_8_flags "-i8 -r8")
 else()
   message(FATAL_ERROR "Unsupported Fortran compiler '${CMAKE_Fortran_COMPILER_ID}'")
 endif()
